beps/bal-build/2200_generating_api_spec.md

```
@@ -0,0 +1,164 @@
+
+```shell 
+--export-service-spec
+```
+
+To extract core service details such as base paths, ports, and protocols, this proposal introduces a new compiler plugin along with compile-time specification generation support in the standard libraries of each protocol. These components are activated when the new flag is provided to the bal build command. 
+
+The compiler plugin introduced, will analyze the Ballerina syntax tree and semantic model to identify service declarations and listeners. The extraction logic will rely on the ballerina compiler API (for syntax tree nodes and semantic analysis) to ensure accuracy and consistency with the actual compiled program.
+
+In addition, protocol-specific standard libraries (e.g., GraphQL, gRPC) will be enhanced to generate the relevant specifications at compile time, such as GraphQL schemas or gRPC proto definitions, so they can be included in the build output. 
+
+ ### Endpoint configuration information extraction
```

markdownlint-cli2 (0.20.0)

[warning] Headings must start at the beginning of the line

(MD023, heading-start-left)